From a90cf6cd0db8799806c48ccc152aef7dfa9cd9c5 Mon Sep 17 00:00:00 2001
From: Zhu,Tianyang <tianyang.zhu@intel.com>
Date: Mon, 1 Jul 2013 19:08:14 +0800
Subject: HDMI-MW: Pass video session ID from MDS(HDMI middleware) to HWC

BZ: 116726

1: Bit 24 ~ Bit 27 and Bit 31 are used by Video playback
2: Add a flag to indicate current layer has video session ID

Category: feature differentiation
Domain: Video.Rendering-MDS
Origin: internal
Upstream-Candidate: no

Change-Id: I00276e85cc3de46b4c2c59c4b4f97bb5d782d1a0
Orig-Change-Id: If3ca1d9dc14c490d42bc47a3afe79132c1491aef
Signed-off-by: Zhu,Tianyang <tianyang.zhu@intel.com>
Signed-off-by: Jason Hu <jason.hu@intel.com>
Signed-off-by: Lin Xie <lin.xie@intel.com>
---
 include/hardware/gralloc.h         |    5 +++++
 include/hardware/hwcomposer_defs.h |    9 ++++++++-
 2 files changed, 13 insertions(+), 1 deletions(-)

diff --git a/include/hardware/gralloc.h b/include/hardware/gralloc.h
index 0dbebcf..dc41ecc 100644
--- a/include/hardware/gralloc.h
+++ b/include/hardware/gralloc.h
@@ -120,10 +120,15 @@ enum {
      */
     GRALLOC_USAGE_PROTECTED             = 0x00004000,
 
+    /* bit 24 ~ bit 27 used by video playback if bit 31 is set */
+    /* mask for the HDMI middleware(MDS) usage bit-mask */
+    GRALLOC_USAGE_MDS_SESSION_ID_MASK   = 0x0F000000,
+
     /* implementation-specific private usage flags */
     GRALLOC_USAGE_PRIVATE_0             = 0x10000000,
     GRALLOC_USAGE_PRIVATE_1             = 0x20000000,
     GRALLOC_USAGE_PRIVATE_2             = 0x40000000,
+    /* used by video playback */
     GRALLOC_USAGE_PRIVATE_3             = 0x80000000,
     GRALLOC_USAGE_PRIVATE_MASK          = 0xF0000000,
 };
diff --git a/include/hardware/hwcomposer_defs.h b/include/hardware/hwcomposer_defs.h
index c800198..7ce144c 100644
--- a/include/hardware/hwcomposer_defs.h
+++ b/include/hardware/hwcomposer_defs.h
@@ -84,10 +84,17 @@ enum {
     HWC_SKIP_LAYER = 0x00000001,
 
     /*
+     * HWC_HAS_VIDEO_SESSION_ID indicates the layer has a video session ID
+     * which is allocated by MDS(HDMI Middleware)
+     * Bit 24 ~ 27 are used too
+     */
+    HWC_HAS_VIDEO_SESSION_ID = 0x20000000,
+
+    /*
      * HWC_TRICK_MODE indicates the layer needs to be rendered on the
      * display device. Currently it is used on hdmi extend video mode.
      */
-    HWC_TRICK_MODE = 0x10000000,
+    HWC_TRICK_MODE = 0x40000000,
 };
 
 /*
-- 
1.7.4.1

